client/lib/posts/utils.js

```diff
@@ -49,6 +49,12 @@ var utils = {
 			if ( site.options.is_mapped_domain ) {
 				previewUrl = previewUrl.replace( site.URL, site.options.unmapped_url );
 			}
+			if ( site.options.frame_nonce ) {
+				parsed = url.parse( previewUrl, true );
+				parsed.query['frame-nonce'] = site.options.frame_nonce;
+				delete parsed.search;
+				previewUrl = url.format( parsed );
+			}
```

Comment:
This function desperately needs unit tests, though understandably non-existent at the moment. I might suggest that you consider adding one to the existing [`utils` test suite](https://github.com/Automattic/wp-calypso/blob/master/client/lib/posts/test/utils.js):

```js
describe( '#getPreviewURL', function() {
	it( 'should include a frame nonce when post has site defined', function() {
		// ...
	} );
} );
```

Not a blocker, but we should be more diligent about testing.